File: CS/XtraGridExportViewAsPDF/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;
using DevExpress.XtraGrid.Views.Grid;

namespace XtraGridExportViewAsPDF
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nwindDataSet.Orders' table. You can move, or remove it, as needed.
            this.ordersTableAdapter.Fill(this.nwindDataSet.Orders);
            
        }

        private void button1_Click(object sender, EventArgs e)
        {
            GridView View = gridControl1.MainView as GridView;
            if (View != null)
            {
                View.ExportToPdf("MainViewData.pdf");

                // The following requires Adobe Reader to be installed on the system
                Process pdfExport = new Process();
                pdfExport.StartInfo.FileName = "AcroRd32.exe";
                pdfExport.StartInfo.Arguments = "MainViewData.pdf";
                pdfExport.Start();
            }
        }
    }
}

File: VB/XtraGridExportViewAsPDF.VB/Form1.vb
Imports System.Diagnostics
Imports DevExpress.XtraGrid.Views.Grid

Public Class Form1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'TODO: This line of code loads data into the 'NwindDataSet.Orders' table. You can move, or remove it, as needed.
        Me.OrdersTableAdapter.Fill(Me.NwindDataSet.Orders)

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Dim View As GridView = TryCast(GridControl1.MainView, GridView)
        If Not (View Is Nothing) Then
            View.ExportToPdf("MainViewData.pdf")

            ' The following requires Adobe Reader to be installed on the system
            Dim pdfExport As New Process()
            pdfExport.StartInfo.FileName = "AcroRd32.exe"
            pdfExport.StartInfo.Arguments = "MainViewData.pdf"
            pdfExport.Start()
        End If
    End Sub
End Class
